ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -14046,7 +14046,8 @@ private void addDbAndTabToOutputs(String[] qualifiedTabName, TableType type,
       t.setSerdeParam(serdeMap.getKey(), serdeMap.getValue());
     }
     WriteType lockType = tblProps != null && Boolean.parseBoolean(tblProps.get(TABLE_IS_CTAS))
-        && AcidUtils.isExclusiveCTASEnabled(conf) ?
+        && AcidUtils.isExclusiveCTASEnabled(conf)
+        && (t.getStorageHandler() == null || !t.getStorageHandler().directInsertCTAS()) ?
```

Comment:
could we please add a comment that with this construct we exclude the iceberg?

Reply:
done